subsys/usb/device/usb_device.c

```diff
@@ -688,6 +719,56 @@ static bool usb_set_configuration(struct usb_setup_packet *setup)
 	return found;
 }
 
+/*
+ * @brief Get 32-bit endpoint bit-mask for specified iface/alt_setting
+ *
+ * Iterate through the USB interface descriptor and accumulate
+ * a bit-mask representing all the endpoints used by by the specified
+ * interface and alternate-setting combination. OUT endpoints are
+ * represented by the lower 16 bit-positions, while the upper 16
+ * bit-positions indicate IN endpoints
+ *
+ * @param [in] iface        The USB interface of interest
+ *        [in] alt_setting  The alternate setting of above interface for
+ *                          which endpoint bit-mask should be accumulated
+ *
+ * @return The accumulated 32-bit endpoint bit-mask
+ */
+static uint32_t get_ep_bm_for_alt_setting(uint8_t iface, uint8_t alt_setting)
+{
+	const uint8_t *p = usb_dev.descriptors;
+	struct usb_ep_descriptor *ep = NULL;
+	uint8_t cur_alt_setting = 0xff;
```

Comment:
The reason behind the magic numbers is to skip endpoint descriptors that are not within requested interface alternate setting. The values `0xff` are extremely unlikely, but are not out of range. I think the purpose behind the variables could be clearer if both `cur_alt_setting` and `cur_iface` were replaced with `bool alt_setting_match = false;`. Then `alt_setting_match` would be set to `true` if descriptor bInterfaceNumber/bAlternateSetting matches requested values (and to `false` on mismatch) instead of assigning the `cur_alt_setting` and `cur_iface`.

Reply:
Thanks - changed